addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/HeaderBasedSessionManagementMethodType.java

```diff
@@ -0,0 +1,454 @@
+
+        method.setHeaderConfigs(mapToList(map));
+
+        return method;
+    }
+
+    @Override
+    public void persistMethodToSession(
+            Session session, int contextId, SessionManagementMethod method)
+            throws UnsupportedSessionManagementMethodException, DatabaseException {
+
+        if (!(method instanceof HeaderBasedSessionManagementMethod)) {
+            throw new UnsupportedSessionManagementMethodException(
+                    "Header based session management type only supports: "
+                            + HeaderBasedSessionManagementMethod.class.getName());
+        }
+        HeaderBasedSessionManagementMethod sessionMethod =
+                (HeaderBasedSessionManagementMethod) method;
+
+        session.setContextData(
+                contextId,
+                RecordContext.TYPE_SESSION_MANAGEMENT_FIELD_1,
+                EncodingUtils.mapToString(listToMap(sessionMethod.getHeaderConfigs())));
+    }
+
+    @Override
+    public void exportData(Configuration config, SessionManagementMethod sessionMethod) {
+        if (!(sessionMethod instanceof HeaderBasedSessionManagementMethod)) {
+            throw new UnsupportedSessionManagementMethodException(
+                    "Header based session management type only supports: "
+                            + HeaderBasedSessionManagementMethod.class.getName());
+        }
+        HeaderBasedSessionManagementMethod method =
+                (HeaderBasedSessionManagementMethod) sessionMethod;
+
+        List<String> list = new ArrayList<>();
+        for (Pair<String, String> pair : method.headerConfigs) {
+            list.add(pair.first + ":" + pair.second);
+        }
+        config.setProperty(CONTEXT_CONFIG_SESSION_MGMT_HEADER, list);
+    }
+
+    @Override
+    public void importData(Configuration config, SessionManagementMethod sessionMethod)
+            throws ConfigurationException {
+        if (!(sessionMethod instanceof HeaderBasedSessionManagementMethod)) {
+            throw new UnsupportedSessionManagementMethodException(
+                    "Header based session management type only supports: "
+                            + HeaderBasedSessionManagementMethod.class.getName());
+        }
+        HeaderBasedSessionManagementMethod method =
+                (HeaderBasedSessionManagementMethod) sessionMethod;
+
+        List<Pair<String, String>> headerConfigs = new ArrayList<>();
+        for (Object entry : config.getList(CONTEXT_CONFIG_SESSION_MGMT_HEADER)) {
+            String str = entry.toString();
+            int cIndex = str.indexOf(":");
+            if (cIndex > 0) {
+                headerConfigs.add(new Pair<>(str.substring(0, cIndex), str.substring(cIndex + 1)));
+            }
+        }
+        method.setHeaderConfigs(headerConfigs);
+    }
+
+    @Override
+    public ApiDynamicActionImplementor getSetMethodForContextApiAction() {
+        return new ApiDynamicActionImplementor(API_METHOD_NAME, null, null) {
```

Comment:
This does not seem to be defining any parameters for the headers/values. Was that on purpose?

Reply:
Er, no :P